pkg/network/protocols/http/protocol.go

```diff
@@ -0,0 +1,192 @@
+			Key:           uint32(protocols.ProgramHTTP),
+			ProbeIdentificationPair: manager.ProbeIdentificationPair{
+				EBPFFuncName: "socket__http_filter",
+			},
+		},
+	},
+}
+
+func newHttpProtocol(cfg *config.Config) (protocols.Protocol, error) {
+	if !cfg.EnableHTTPMonitoring {
+		return nil, nil
+	}
+
+	kversion, err := kernel.HostVersion()
+	if err != nil {
+		return nil, fmt.Errorf("couldn't determine current kernel version: %w", err)
+	}
+
+	if kversion < MinimumKernelVersion {
+		return nil, fmt.Errorf("http feature not available on pre %s kernels", MinimumKernelVersion.String())
+
+	}
+
+	telemetry := NewTelemetry()
+
+	return &httpProtocol{
+		cfg:       cfg,
+		telemetry: telemetry,
+	}, nil
+}
+
+// ConfigureOptions add the necessary options for the http monitoring to work,
+// to be used by the manager. These are:
+// - Set the `http_in_flight` map size to the value of the `max_tracked_connection` configuration variable.
+//
+// We also configure the http event stream with the manager and its options.
+func (p *httpProtocol) ConfigureOptions(mgr *manager.Manager, opts *manager.Options) {
+	opts.MapSpecEditors[httpInFlightMap] = manager.MapSpecEditor{
+		Type:       ebpf.Hash,
+		MaxEntries: uint32(p.cfg.MaxTrackedConnections),
+		EditorFlag: manager.EditMaxEntries,
+	}
+
+	// Configure event stream
+	events.Configure("http", mgr, opts)
+}
+
+func (p *httpProtocol) PreStart(mgr *manager.Manager) (err error) {
+	p.eventsConsumer, err = events.NewConsumer(
+		"http",
+		mgr,
+		p.processHTTP,
+	)
+	if err != nil {
+		return
+	}
+
+	p.statkeeper = NewHTTPStatkeeper(p.cfg, p.telemetry)
+	p.eventsConsumer.Start()
+
+	return
+}
+
+func (p *httpProtocol) PostStart(mgr *manager.Manager) error {
+	// Setup map cleaner after manager start.
+	p.setupMapCleaner(mgr)
+
+	return nil
+}
+
+func (p *httpProtocol) PreStop(mgr *manager.Manager) {
+	// mapCleaner handles nil pointer receivers
+	p.mapCleaner.Stop()
+
+	if p.eventsConsumer != nil {
+		p.eventsConsumer.Stop()
+	}
+
+	if p.statkeeper != nil {
+		p.statkeeper.Close()
+	}
+}
+
+func (p *httpProtocol) PostStop(mgr *manager.Manager) {}
+
+func (p *httpProtocol) DumpMaps(output *strings.Builder, mapName string, currentMap *ebpf.Map) {
+	switch mapName {
+	case httpInFlightMap: // maps/http_in_flight (BPF_MAP_TYPE_HASH), key ConnTuple, value httpTX
+		output.WriteString("Map: '" + mapName + "', key: 'ConnTuple', value: 'httpTX'\n")
+		iter := currentMap.Iterate()
+		var key netebpf.ConnTuple
+		var value HttpTX
+		for iter.Next(unsafe.Pointer(&key), unsafe.Pointer(&value)) {
+			output.WriteString(spew.Sdump(key, value))
+		}
+
+	}
+}
+
+func (p *httpProtocol) processHTTP(data []byte) {
+	tx := (*EbpfHttpTx)(unsafe.Pointer(&data[0]))
+	p.telemetry.Count(tx)
+	p.statkeeper.Process(tx)
+}
+
+func (p *httpProtocol) setupMapCleaner(mgr *manager.Manager) {
+	httpMap, _, err := mgr.GetMap(httpInFlightMap)
+	if err != nil {
+		log.Errorf("error getting http_in_flight map: %s", err)
```

Comment:
```suggestion
		log.Errorf("error getting %s map: %s", httpInFlightMap, err)
		return
```

Reply:
return as the ma is nil, we cannot (and should not) setup the cleaner is such a case